src/inventory_ui.cpp

```diff
@@ -4280,8 +4282,10 @@ bool pickup_selector::wear()
 
 void pickup_selector::add_reopen_activity()
 {
-    u.assign_activity( pickup_menu_activity_actor( where, to_use ) );
-    u.activity.auto_resume = true;
+    // copy the member variables to still be valid on call
+    uistate.open_menu = [where = where, to_use = to_use]() {
+        get_avatar().pick_up( game_menus::inv::pickup( where, to_use ) );
```

Comment:
I think you can use `get_player_character` here instead and avoid including `avatar.h`.
Also renaming the method to something more representative of what it's doing now would be nice.

Reply:
Thanks! I didn't know about `get_player_character`.

I love that we are the same person. I was worried about adding `avatar.h` and did want to rename that method :D.